src/rgw/rgw_sync_module_es.cc

```diff
@@ -167,58 +223,158 @@ struct ElasticConfig {
 
 using ElasticConfigRef = std::shared_ptr<ElasticConfig>;
 
-struct es_dump_type {
-  const char *type;
-  const char *format;
-  bool analyzed;
+static const char*es_type_to_str(const ESType& t) {
```

Comment:
nit spacing

Reply:
I actually realize that the only way default would be triggered would be if we supply a ESType that is not in the type enum (otherwise we get a compile error anyway). Implementing a new type would anyway require both the enum and string representation, so we could probably do away with the nullptr and subsequent handling?

Reply:
Yeah. Can probably just assert() there. Maybe that's not even needed.